### ISSUE TITLE
Dependencies with `importlib.import_module` are not detected

### ISSUE THREAD

**Issue by lmmx** [fpgmaas/deptry#780]
**Describe the bug**

- It is possible to declare dependencies using `importlib.import_module("foo")` rather than `import foo`
- These are not identified by `deptry`

**To Reproduce**

- Add `tests/data/some_dyn_imports.py`

```py
from importlib import import_module

import_module("polars")
```

- Add test to `tests/unit/imports/test_extract.py`:

```py
def test_dyn_import_parser_py() -> None:
     some_dyn_imports_path = Path("tests/data/some_dyn_imports.py")

     assert get_imported_modules_from_list_of_files([some_dyn_imports_path]) == {
        "importlib": [Location(some_dyn_imports_path, line=1, column=1)],
        "polars": [Location(some_dyn_imports_path, line=3, column=1)],
     }
```

**Expected behavior**

The test should pass in cases where the import argument is specified.

**System:**

-   OS: e.g. Linux Mint 21
-   Language Version: Python 3.12
-   PDM version: PDM 2.17.1

**Edit** I extended the node visitor to accept nodes of this type, initial version captures

- `import importlib` -> `importlib.import_module("a")` https://github.com/lmmx/deptry/commit/709acb8eccde17e802290dad01916e439ee3914b
- `from importlib import import_module` -> `import_module("b")` https://github.com/lmmx/deptry/commit/025df9734f4699b6025303079f81e917118bd19c
- `from importlib import import_module as im` -> `im("c")` https://github.com/lmmx/deptry/commit/73c790a033666ae7601562cce29626937d7791fe

**Comment by mkniewallner** [comment]
This would be a great addition to the library. Will gladly accept a pull request for this feature.

**Comment by lmmx** [comment]
My pleasure:

- #782

Also added a note in the docs usage page to clarify that this is only for string literals ([code search shows](https://grep.app/search?q=importlib.import_module%28) this is not uncommon in real code)

**Comment by mkniewallner** [comment]
Resolved by #782.
